### ISSUE TITLE
question

### ISSUE THREAD

**Issue by AIikai** [ShannonAI/mrc-for-flat-nested-ner#5]
Hello, excuse me, is this program runnable? I tried several times. It was stopping at the end of the first epoch and not continuing every time. I wonder if there's something wrong.  Thank you.

**Comment by littlesulley** [comment]
yeah. the code is runable. You can reproduce the results listed in the "log" folder.

From your description, I guess the problem might be short of memory. Try using a GPU with larger memory or decrease the value of batch size ?

Or directly paster the error you encountered here ?

**Comment by AIikai** [comment]
if nb_tr_steps % config.checkpoint == 0:
        print("-*-"*15)
        print("current training loss is : ")
        print(loss.item())
        tmp_dev_loss, tmp_dev_acc, tmp_dev_prec, tmp_dev_rec, tmp_dev_f1 = eval_checkpoint(model, dev_dataloader, config, device, n_gpu, label_list, eval_sign="dev")
        print("......"*10)
        print("DEV: loss, acc, precision, recall, f1")
        print(tmp_dev_loss, tmp_dev_acc, tmp_dev_prec, tmp_dev_rec, tmp_dev_f1)
应该不是这个问题，我之前就试过了。程序第一次跑到上面的时候， 进入eval_checkpoint，然后在下面的eval计算f1，recall值前停住，很久很久，一直没动静，不报错也没print。
    if config.entity_sign == "flat":
        eval_accuracy, eval_precision, eval_recall, eval_f1 = flat_ner_performance(start_pred_lst, end_pred_lst, span_pred_lst, start_gold_lst, end_gold_lst, span_gold_lst, ner_cate_lst, label_list, threshold=config.entity_threshold, dims=2)
    else:
        eval_accuracy, eval_precision, eval_recall, eval_f1 = nested_ner_performance(start_pred_lst, end_pred_lst, span_pred_lst, start_gold_lst, end_gold_lst, span_gold_lst, ner_cate_lst, label_list, threshold=config.entity_threshold, dims=2)
    average_loss = round(eval_loss / eval_steps, 4)  
    eval_f1 = round(eval_f1, 4)
    eval_precision = round(eval_precision, 4)
    eval_recall = round(eval_recall, 4)
    eval_accuracy = round(eval_accuracy, 4)
    print('average_loss, eval_accuracy, eval_precision, eval_recall, eval_f1')
    print(average_loss, eval_accuracy, eval_precision, eval_recall, eval_f1)

**Comment by FayolChang** [comment]
这是因为 bmes_dode 在处理 current_label = 'E' 但其之前不是'B'时，会陷入死循环

**Comment by AIikai** [comment]
> 这是因为 bmes_dode 在处理 current_label = 'E' 但其之前不是'B'时，会陷入死循环

厉害了，谢谢大佬！已解决这个问题。

**Comment by lazyFrogLOL** [comment]
> > 这是因为 bmes_dode 在处理 current_label = 'E' 但其之前不是'B'时，会陷入死循环
> 
> 厉害了，谢谢大佬！已解决这个问题。

您好，想问一下是如何解决的？
